### PR TITLE
Parallel projection and discrete zoom

### PR DESCRIPTION
**Pull request by manthey**
Added parallel projection and discrete zoom.

Zoom levels are now consistent with other mapping libraries and are no longer based on the window size.  As such, the zoom level is DIFFERENT from what it was, unless the window was exactly one standard map tile high (256 pixels).

Pass some configuration into the renderers.

Update tests to have better window resizing and to compensate for changes in zoom definition.  This breaks all of the screen shot tests.

Clustering needs to know the size of the layer so that the scale can be appropriate, as the scale is now fixed to the zoom level rather than a combination of zoom level and window size.

Add jasmine test for discrete zoom and parallel projection.  Add selenium test for parallel projection.

Add a little bit of documentation to the map class.

### CODE REVIEW

**Comment by aashish24** [comment]
:clap:

**Comment by jbeezley** [comment]
Yay!

I restarted the buildbot build.  I have seen a couple of errors related to downloads from midas failing the last couple of days.  I think it just some local network issues, hopefully they don't persist.

**Comment by aashish24** [comment]
I am reviewing the VGL branch first and then will get back to this one.

**Comment by jeffbaumes** [comment]
I feel like I will need to get 5 or 6 demos of what's new in GeoJS when I get back in town. Great work @manthey @jbeezley @aashish24 @danlamanna! :tada:

File: src/core/map.js [hunk 1]
```diff
@@ -1,6 +1,26 @@
...
+ *  width:
+ *  height:
+ *  parallelProjection: (bool) - true to use parallel projection, false to use
+ *      perspective.
+ *  discreteZoom: (bool) - true to only allow integer zoom levels, false to
+ *      allow any zoom level.
+ *  autoResize:
+ *  clampBounds:
+ *  interactor:
+ *  clock:
```
Comment (jbeezley) [inline src/core/map.js]: Yes, we should start documenting class options like this.

File: src/core/event.js [hunk 2]
```diff
@@ -184,6 +184,16 @@ geo.event.transitionstart = "geo_transitionstart";
...
 
+//////////////////////////////////////////////////////////////////////////////
+/**
+ * Triggered when the parallel projection mode is changes.
+ *
+ * @property paralellProjection {boolean} True if parallel projection is turned
+ *                                        on.
+ */
+//////////////////////////////////////////////////////////////////////////////
+geo.event.parallelprojection = "geo_parallelprojection";
```
Comment (jbeezley) [inline src/core/event.js]: Will this event change the displayed map bounds?

File: src/util/clustering.js [hunk 3]
```diff
@@ -127,23 +127,26 @@
...
   /**
-   *
    * This class manages clustering of an array of positions hierarchically.
    * The algorithm and code was adapted from the Leaflet marker cluster
    * plugin by David Leaver: https://github.com/Leaflet/Leaflet.markercluster
    *
    * @class geo.util.ClusterGroup
    * @param {object} opts An options object
+   * @param {number} width The width of the window; used for scaling.
+   * @param {number} height The height of the window; used for scaling.
```
Comment (jbeezley) [inline src/util/clustering.js]: I'm not sure about adding the window dimensions here.  What the class needs is the `unitsPerPixel` method to determine the scaling for the clustering distance.  This is perhaps something that I can fix with the new camera API.

File: src/core/event.js (hunk 2 above)
Comment (manthey) [inline src/core/event.js]: Not currently.  If we ever implement a tilted or rotated view then it will.

File: .gitignore [hunk 4]
```diff
@@ -1,4 +1,6 @@
 /build/*
+/_build/*
+/Testing/*
```
Comment (jbeezley) [inline .gitignore]: My filesystem isn't case sensitive, so this ignores all files under `testing`.   :(

**Comment by aashish24** [comment]
@jbeezley I didn't have any major questions here. Feel free to merge it if it looks good to you.

**Comment by aashish24** [comment]
@manthey Great stuff.. :clap:

**Comment by jbeezley** [comment]
As you noted, the discrete zoom feature doesn't work well with touch events, but we can treat it as an experimental feature and open a separate issue for it.

[LGTM](http://lgtm.in/i/wDB1)
Ship it!

**Comment by manthey** [comment]
Thanks, @jbeezley for dealing with the Selenium tests.

**Comment by jbeezley** [comment]
Penance for designing them.

**Comment by aashish24** [comment]
@jbeezley nice approval image! .... :+1:

**Pull request merged**